cmd/ci-operator/main.go

```diff
@@ -2243,3 +2257,38 @@ func addSchemes() error {
 	}
 	return nil
 }
+
+// getClusterProfileSecret retrieves the cluster profile secret name using config resolver,
+// and gets the secret from the ci namespace
+func getClusterProfileSecret(clusterProfile string, client ctrlruntimeclient.Client, resolverClient server.ResolverClient, ctx context.Context) (*coreapi.Secret, error) {
+	// Use config-resolver to get details about the cluster profile (which includes the secret's name)
+	cpDetails, err := resolverClient.ClusterProfile(clusterProfile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve details from config resolver for '%s' cluster profile", clusterProfile)
+	}
+	// Get the secret from the ci namespace. We expect it exists
+	ciSecret := &coreapi.Secret{}
+	err = client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: "ci", Name: cpDetails.Secret}, ciSecret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get secret '%s' from ci namespace: %w", cpDetails.Secret, err)
+	}
+
+	return ciSecret, nil
+}
+
+// getClusterProfileNamesFromTargets extracts the needed cluster profile name(s) from the target arg(s)
+func (o *options) getClusterProfileNamesFromTargets() {
+	for _, targetName := range o.targets.values {
+		for _, test := range o.configSpec.Tests {
```

Comment:
Consider having a helper method on the Tests object to retrieve the cluster profiles.

Reply:
You can add a method to the `Tests` corresponding struct to operator on the cluster profiles

Reply:
check also https://github.com/psalajova/ci-tools/blob/2a6fab78c976f5188ae62af7a5e7ec51ddf5de78/pkg/api/types.go#L800-L806 for example